### ISSUE TITLE
ESM build has classes downleveled to functions, but CJS build does not

### ISSUE THREAD

**Issue by bradzacher** [js-sdsl/js-sdsl#158]
See files published on NPM:
ESM: https://unpkg.com/browse/js-sdsl@4.3.0/dist/esm/container/OtherContainer/Queue.js
CJS: https://unpkg.com/browse/js-sdsl@4.3.0/dist/cjs/container/OtherContainer/Queue.js

If I had to guess it's because the ESM build is set to target ES5:
https://github.com/js-sdsl/js-sdsl/blob/28fe66c6cd2356b9b849ff0cd771050b679c3d32/gulpfile.ts#L27-L34

**Comment by ZLY201** [comment]
Yes, We hope that Js-sdsl can be compatible with more browsers so we set target to ES5 for ES module.

**Comment by bradzacher** [comment]
All modern browsers support much higher than es5 though - esp if they truly support ESM.
For example every modern browser supports es6 classes https://caniuse.com/es6-class 

Why not just leave it to the user to downlevel your code for their target environment? It's going to be very, very rare that anyone is going to consume your package on the web without passing it through a bundler first.

The issue is that anyone consuming your package from nodejs is stuck using your downleveled code if they import your package via ESM - which is a really backwards experience!

**Comment by ZLY201** [comment]
> All modern browsers support much higher than es5 though - esp if they truly support ESM. For example every modern browser supports es6 classes https://caniuse.com/es6-class
> 
> Why not just leave it to the user to downlevel your code for their target environment? It's going to be very, very rare that anyone is going to consume your package on the web without passing it through a bundler first.
> 
> The issue is that anyone consuming your package from nodejs is stuck using your downleveled code if they import your package via ESM - which is a really backwards experience!

Sounds good, we might change the building target of ESM to ES6 at version 5.0.
